server/src/main/java/org/jboss/as/server/operations/AbstractInstallationReporter.java

```diff
@@ -0,0 +1,303 @@
+    private ModelNode createCPUNode() throws OperationFailedException {
+        ModelNode cpu = new ModelNode().setEmptyObject();
+        cpu.get(ARCH).set(getProperty("os.arch"));
+        cpu.get(AVAILABLE_PROCESSORS).set(Runtime.getRuntime().availableProcessors());
+        return cpu;
+    }
+
+    /**
+     * Create a ModelNode representing the instance product and the plateform it is running on.
+     *
+     * @param context the operation context.
+     * @param environment the current runtime environment.
+     * @param patchingInfo
+     * @param installationDir installation directory of the current server instance.
+     * @return a ModelNode representing the instance product and the plateform it is running on.
+     * @throws OperationFailedException
+     */
+    protected ModelNode createProductNode(OperationContext context, ProductEnvironment environment,
+            ModelNode patchingInfo, Path installationDir) throws OperationFailedException {
+        assert environment != null;
+        ModelNode product = new ModelNode().setEmptyObject();
+        product.get(REPORT_VERSION).set("1.0");
+        product.get(NODE_NAME).set(environment.getHostName());
+        product.get(HOSTNAME).set(environment.getQualifiedHostName());
+        product.get(INSTANCE_ID).set(environment.getInstanceUuid().toString());
+        PathAddress organizationAddress = PathAddress.EMPTY_ADDRESS;
+        if(context.getProcessType() == ProcessType.HOST_CONTROLLER) {
+            organizationAddress = PathAddress.pathAddress(HOST, environment.getHostControllerName());
+        }
+        ModelNode root = context.readResourceFromRoot(organizationAddress, false).getModel();
+        if(root.hasDefined(ORGANIZATION)) {
+            product.get(ORGANIZATION).set(root.get(ORGANIZATION));
+        }
+        if (environment.getProductConfig() != null) {
+            if(environment.getProductConfig().getProductName() != null) {
+                product.get(PRODUCT_NAME).set(environment.getProductConfig().getProductName());
+            }
+            if(environment.getProductConfig().getProductVersion() != null) {
+                product.get(PRODUCT_VERSION).set(environment.getProductConfig().getProductVersion());
+            }
+            if (environment.getProductConfig().isProduct()) {
+                product.get(PRODUCT_COMMUNITY_IDENTIFIER).set(PRODUCT_TYPE);
+            } else {
+                product.get(PRODUCT_COMMUNITY_IDENTIFIER).set(PROJECT_TYPE);
+            }
+        }
+        if(installationDir != null && Files.exists(installationDir)) {
+            product.get(PRODUCT_HOME).set(installationDir.toAbsolutePath().toString());
+            BasicFileAttributeView view = Files.getFileAttributeView(installationDir, BasicFileAttributeView.class);
+            try {
+                BasicFileAttributes attributes = view.readAttributes();
+                product.get(PRODUCT_INSTALLATION_DATE).set(attributes.creationTime().toString());
```

Comment:
This may not be accurate. Per the javadoc for it, some filesystems may report last-modified-time instead, others may report something like the epoch.

Probably not worth worrying about though. Only thing I could think of is checking if attributes.creationTime().toMillis() == 0 and not reporting that.